website/static/js/fangorn.js

```diff
@@ -1009,6 +1020,42 @@ function _removeEvent (event, items, col) {
     }
 }
 
+function doCheckout(item, checkout, showError) {
+    var ret = $.Deferred();
+    var request = $osf.ajaxJSON(
+        'PUT',
+        window.contextVars.apiV2Prefix + 'files' + item.data.path + '/',
+        {
+            isCors: true,
+            data: {
+                data: {
+                    id: item.data.path.replace('/', ''),
+                    type: 'files',
+                    attributes: {
+                        checkout: checkout
+                    }
+                }
+            }
+        }
+    );
+    request.done(function(resp) {
+        if (showError) {
+            window.location.reload();
+        }
+        ret.resolve(resp);
+    });
+    request.fail(function(xhr, status, error) {
+        if (showError) {
+            $osf.growl('Error', 'Unable to check-out file. This is most likely due to the file being already checked-out' +
+                ' by another user.');
+        }
+        ret.reject(xhr, status, error);
```

Comment:
Calling reject in the jqxhr's` fail and `resolve` method in its `done` method is exactly equivalent to returning the `jqxhr` itself. Keep in mind that jqxhrs are themselves Promises.